src/TwitterDB.js

```diff
@@ -0,0 +1,462 @@
+    if(err){
+      log.error("TwitDB","Error opening database, %s");
+    }
+  });
+}
+
+
+/**
+ * TwitterDB.prototype.init - Checks the database has all the tables needed.
+ * @function
+ */
+TwitterDB.prototype.init = function() {
+  log.info("TwitDB","Starting DB Init");
+  this._create_profile_cache();
+  this._create_twitter_table();
+  this._create_dm_room_table();
+  this._create_timeline_table();
+}
+
+TwitterDB.prototype.get_profile_by_id = function(id){
+  log.info("TwitDB","Retrieving profile: %s",id);
+  return new Promise((resolve,reject) =>{
+    this.db.get(
+      `
+      SELECT profile, timestamp
+      FROM user_cache
+      WHERE user_cache.id = $id;
+      `
+    ,{
+      $id: id
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving profile: %s",err.Error);
+        reject(err);
+      }
+      if(row !== undefined){
+        row.profile = JSON.parse(row.profile);
+        resolve(row);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+/**
+ @typedef TwitterDBProfile
+ @type {Object}
+ @property {object} profile   A complete record of a twitter users profile.
+ @property {number} timestamp The last time the record was updated.
+ */
+
+/**
+ * TwitterDB.prototype.get_profile_by_name - Get a Twitter profile by the
+ * screenname of a user.
+ * @function
+ * @returns {Promise<TwitterDBProfile>}
+ */
+TwitterDB.prototype.get_profile_by_name = function(name){
+  return new Promise((resolve,reject) => {
+    this.db.get(
+      `
+      SELECT profile, timestamp
+      FROM user_cache
+      WHERE user_cache.screenname = $name;
+      `
+    ,{
+      $name: name
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving profile: %s",err.Error);
+        reject(err);
+        return;
+      }
+      if(row !== undefined){
+        row.profile = JSON.parse(row.profile);
+        resolve(row);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+/**
+ * TwitterDB.prototype.set_twitter_profile - Insert/Update a Twitter profile
+ * into the database.
+ * @function
+ * @param  {type} id          Twitter ID of the profile.
+ * @param  {string} name      screenname of the profile.
+ * @param  {object} data      The profile data.
+ * @param  {number} timestamp The time when this data was *fetched*.
+ */
+TwitterDB.prototype.set_twitter_profile = function(id,name,data,timestamp){
+  this.db.run(
+    `
+    REPLACE INTO user_cache VALUES ($id,$name,$data,$timestamp);
+    `
+  ,{
+    $id: id,
+    $name: name,
+    $data: JSON.stringify(data),
+    $timestamp: timestamp
+  },
+  function (err) {
+    if(err){
+      log.error("TwitDB","Error storing profile: %s",err);
+      return;
+    }
+  });
+}
+
+TwitterDB.prototype.get_client_data = function(user_id){
+  return new Promise((resolve,reject) =>{
+    this.db.get(
+      `
+      SELECT *
+      FROM twitter_account
```

Comment:
Terminology. A function called `get_client_data` I would expect to pull stuff from a `client_data` or maybe `clients` table, but no, it's pulling from `twitter_account`, *obviously*.